pkg/resource/deploy/step_generator.go

```diff
@@ -853,6 +853,103 @@ func processIgnoreChanges(inputs, oldInputs resource.PropertyMap,
 	return ignoredInputs.ObjectValue(), nil
 }
 
+func copyWithUnknownElements(oldValue resource.PropertyValue) resource.PropertyValue {
+	unknown := resource.MakeComputed(resource.NewStringProperty(""))
+	switch {
+	case oldValue.IsObject():
+		newValue := resource.PropertyMap{}
+		for k := range oldValue.ObjectValue() {
+			newValue[k] = unknown
+		}
+		return resource.NewObjectProperty(newValue)
+	case oldValue.IsArray():
+		newValue := make([]resource.PropertyValue, len(oldValue.ArrayValue()))
+		for i := range newValue {
+			newValue[i] = unknown
+		}
+		return resource.NewArrayProperty(newValue)
+	default:
+		return unknown
+	}
+}
+
+func copyStableValue(oldState, newState resource.PropertyMap, path resource.PropertyPath, allowPartialValues bool) {
+	if len(path) == 0 || len(path) > 1 && !allowPartialValues {
+		return
+	}
+
+	oldParent, newParent := resource.NewObjectProperty(oldState), resource.NewObjectProperty(newState)
+	for _, element := range path[:len(path)-1] {
+		switch element := element.(type) {
+		case string:
+			if !oldParent.IsObject() || !newParent.IsObject() {
+				return
+			}
+			key := resource.PropertyKey(element)
+			old, ok := oldParent.ObjectValue()[key]
+			if !ok {
+				return
+			}
+			new, ok := newParent.ObjectValue()[key]
+			if !ok || new.IsComputed() {
+				new = copyWithUnknownElements(old)
+				newParent.ObjectValue()[key] = new
+			}
+			oldParent, newParent = old, new
+		case int:
+			if !oldParent.IsArray() || !newParent.IsArray() {
+				return
+			}
+			oldArray, newArray := oldParent.ArrayValue(), newParent.ArrayValue()
+			if element < 0 || element >= len(oldArray) || element >= len(newArray) {
+				return
+			}
+			old, new := oldArray[element], newArray[element]
+			if new.IsComputed() {
+				new = copyWithUnknownElements(old)
+				newArray[element] = new
+			}
+			oldParent, newParent = old, new
+		}
+	}
+
+	switch element := path[len(path)-1].(type) {
```

Comment:
clearly you're handling the last piece slightly differently.  i'm just not sure why.  could use some comments in this code to help clear up cases.